File: hw/tipantiza/u3/hw22MazeSolidPrinciples/MazeSystem_java/MazeSystem/src/main/java/ec/edu/espe/mazesystem/generator/MazeGenerator.java
package ec.edu.espe.mazesystem.generator; 

import ec.edu.espe.mazesystem.model.AbstractMaze;
import ec.edu.espe.mazesystem.model.Door;
import ec.edu.espe.mazesystem.model.Path;
import ec.edu.espe.mazesystem.model.Room;

import java.util.Random;

public class MazeGenerator implements MazeGeneratorAlgorithm {
    private Random random;

    public MazeGenerator() {
        this.random = new Random();
    }

    @Override
    public void generateMaze(AbstractMaze maze) {
        for (int y = 0; y < maze.getHeight(); y++) {
            for (int x = 0; x < maze.getWidth(); x++) {
                Room room = new Room(x, y);
                
                if (random.nextBoolean()) {
                    room.addDoor(new Door());
                }
                if (random.nextBoolean()) {
                    room.addDoor(new Door(true));
                }
                
                maze.addRoom(x, y, room);
            }
        }

        Room entrance = maze.getRoom(0, 0);
        if (entrance != null) {
            entrance.setEntrance(true);
        }

        Room exit = maze.getRoom(maze.getWidth() - 1, maze.getHeight() - 1);
        if (exit != null) {
            exit.setExit(true);
        }

        Path path = new Path();
        for (int y = 0; y < maze.getHeight(); y++) {
            for (int x = 0; x < maze.getWidth(); x++) {
                Room room = maze.getRoom(x, y);
                if (room != null) {
                    path.addRoom(room);
                }
            }
        }
        maze.setSolutionPath(path);
    }
}

File: hw/vizcaino/u3/hw22SOLIDMaze/hw22Python/generator/maze_generator.py
import random

from generator.maze_generator_algorithm import MazeGeneratorAlgorithm
from model.door import Door
from model.path import Path
from model.room import Room


class MazeGenerator(MazeGeneratorAlgorithm):

    def __init__(self):
        self.random = random.Random()

    def generate_maze(self, maze):

        for y in range(maze.get_height()):
            for x in range(maze.get_width()):

                room = Room(x, y)

                if self.random.choice([True, False]):
                    room.add_door(Door())

                if self.random.choice([True, False]):
                    room.add_door(Door(True))

                maze.add_room(x, y, room)

        entrance = maze.get_room(0, 0)

        if entrance is not None:
            entrance.set_entrance(True)

        exit_room = maze.get_room(
            maze.get_width() - 1,
            maze.get_height() - 1
        )

        if exit_room is not None:
            exit_room.set_exit(True)

        path = Path()

        for y in range(maze.get_height()):
            for x in range(maze.get_width()):

                room = maze.get_room(x, y)

                if room is not None:
                    path.add_room(room)

        maze.set_solution_path(path)